api/oc_network_events.c

```diff
@@ -91,6 +93,17 @@ OC_PROCESS_THREAD(oc_network_events, ev, data)
 void
 oc_network_receive_event(oc_message_t *message)
 {
+  if (!message) {
+    return;
+  }
+  // validate UDP/DTLS message, TCP messages is validated directly in the port
+  // layer for closing the connection
+  if (((message->endpoint.flags & TCP) == 0) &&
+      !oc_udp_is_valid_header(message, &message->endpoint)) {
+    OC_ERR("invalid header - dropping message");
+    oc_message_unref(message);
+    return;
+  }
   if (!oc_process_is_running(&oc_network_events)) {
```

Comment:
I would move this check before checking of the header (line 99)

Reply:
fixed